### ISSUE TITLE
Adding StyleAligned

### ISSUE THREAD

**Issue by shermansiu** [huggingface/diffusers#6063]
### Model/Pipeline/Scheduler description

StyleAligned allows images to be created that follow the style of a reference image, without fine-tuning the pre-trained diffusion model! The method uses adaptive instance normalization layers and with the exception of the reference image, all other images to be generated perform a shared self-attention on the reference image.

![image](https://github.com/huggingface/diffusers/assets/12627125/c3aa53f3-4249-409a-9b9b-43efb6ed0592)

### Open source status

- [X] The model implementation is available.
- [ ] The model weights are available (Only relevant if addition is not a scheduler).

### Provide useful links for the implementation

Project page: https://style-aligned-gen.github.io/
Paper: https://style-aligned-gen.github.io/data/StyleAligned.pdf
Code: https://github.com/google/style-aligned
@amirhertz

**Comment by a-r-r-o-w** [comment]
This looks very cool @amirhertz! Awesome work by you and your co-authors. I'd love to try and understand your paper in the next few days :heart: 

I made a small hacky diffusers-like pipeline using your code for anyone wanting to try it: [Colab](https://colab.research.google.com/drive/15X2E0jFPTajUIjS0FzX50OaHsCbP2lQ0?usp=sharing). Will work on making it more generic to work with any SD pipeline.

<details>
<summary>Code</summary>

```py
from diffusers import DDIMScheduler

scheduler = DDIMScheduler(
    beta_start=0.00085,
    beta_end=0.012,
    beta_schedule="scaled_linear",
    clip_sample=False,
    set_alpha_to_one=False
)
pipe = StyleAlignedPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0",
    torch_dtype=torch.float16,
    variant="fp16",
    use_safetensors=True,
    scheduler=scheduler
).to("cuda")

style_aligned_args = StyleAlignedArgs(
    share_group_norm=False,
    share_layer_norm=False,
    share_attention=True,
    adain_queries=True,
    adain_keys=True,
    adain_values=False,
)
pipe.register_style_aligned(style_aligned_args)

prompt = [
    "a toy train. macro photo. 3d game asset",
    "a toy airplane. macro photo. 3d game asset",
]

images = pipe(
    prompt=prompt,
    negative_prompt="low quality, disfigured",
    num_inference_steps=20,
).images
```
</details>

**Comment by a-r-r-o-w** [comment]
@sayakpaul @DN6 @patrickvonplaten This was a very cool paper and simple enough that someone newer to the field, like me, could understand it easily. Would it make a good addition into community pipelines/research projects? It requires adding the novel Shared Attention processor introduced in the paper, and can be used with txt2img, img2img, inpaint, controlnet, multidiffusion, etc. quite easily.

**Comment by sayakpaul** [comment]
Feel free to add to the community folder!
